ibind/support/oauth.py

```diff
@@ -0,0 +1,340 @@
+    dh_response = result.data["diffie_hellman_response"]
+    lst_signature = result.data["live_session_token_signature"]
+    live_session_token = calculate_live_session_token(
+        dh_prime=dh_prime,
+        dh_random_value=dh_random,
+        dh_response=dh_response,
+        prepend=prepend
+    )
+
+    if not validate_live_session_token(
+            live_session_token=live_session_token,
+            live_session_token_signature=lst_signature,
+            consumer_key=var.IBIND_CONSUMER_KEY
+    ):
+        raise Exception("Live session token validation failed.")
+
+    return live_session_token, lst_expires
+
+
+def prepare_oauth():
+    dh_prime = pem_to_dh_prime(pem_file_path=var.IBIND_DH_PRIME_FP)
+
+    dh_random = generate_dh_random_bytes()
+    dh_challenge = generate_dh_challenge(
+        dh_prime=dh_prime,
+        dh_generator=int(var.IBIND_DH_GENERATOR),
+        dh_random=dh_random,
+    )
+    prepend = calculate_live_session_token_prepend(
+        access_token_secret=var.IBIND_ACCESS_TOKEN_SECRET,
+        private_encryption_key=read_private_key(private_key_fp=var.IBIND_ENCRYPTION_KEY_FP)
+    )
+
+    extra_headers = {"diffie_hellman_challenge": dh_challenge}
+
+    return prepend, extra_headers, dh_prime, dh_random
+
+
+def generate_oauth_headers(
+        request_method: str,
+        request_url: str,
+        live_session_token: Optional[str] = None,
+        extra_headers: Optional[dict[str, str]] = None,
+        request_params: dict = None,
+        signature_method: str = "HMAC-SHA256",
+        prepend: Optional[str] = None,
+):
+    headers = {
+        "oauth_consumer_key": var.IBIND_CONSUMER_KEY,
+        "oauth_nonce": generate_oauth_nonce(),
+        "oauth_signature_method": signature_method,
+        "oauth_timestamp": generate_request_timestamp(),
+        "oauth_token": var.IBIND_ACCESS_TOKEN
+    }
+
+    if var.IBIND_ACCESS_TOKEN:
+        headers.update({"oauth_token": var.IBIND_ACCESS_TOKEN})
+    if extra_headers:
+        headers.update(extra_headers)
+
+    base_string = generate_base_string(
+        request_method=request_method,
+        request_url=request_url,
+        request_headers=headers,
+        request_params=request_params,
+        prepend=prepend
+    )
+
+    if signature_method == "HMAC-SHA256":
+        signature = generate_hmac_sha_256_signature(
+            base_string=base_string,
+            live_session_token=live_session_token)
+    else:
+        private_signature_key = read_private_key(var.IBIND_SIGNATURE_KEY_FP)
+        signature = generate_rsa_sha_256_signature(
+            base_string=base_string,
+            private_signature_key=private_signature_key)
+
+    headers.update({"oauth_signature": signature})
+    headers_string = generate_authorization_header_string(
+        request_data=headers,
+        realm=var.IBIND_REALM
+    )
+
+    header_oauth = {"Authorization": headers_string}
+
+    return header_oauth
+
+
+def pem_to_dh_prime(pem_file_path):
```

Comment:
I've noticed that you read this from a file instead of passing it as an environment variable like IBKR's examples do. Any reason for it?